src/Utils/Validator/Validator.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Indodana\Utils\Validator;
+
+class Validator
+{
+    private static $errors = [];
+
+    private static $input = [];
+
+    public function __construct($input)
+    {
+        self::$input = $input;
+    }
+
+    public static function create($input) {
+        self::$input = [];
+        self::$errors = [];
+
+        return new Validator($input);
+    }
+
+    public function isSuccess()
+    {
+        return empty(self::$errors);
+    }
+
+    public function getErrorMessages()
+    {
+        return self::$errors;
+    }
+
+    public function printErrorMessages()
+    {
+        return join(", ", $this->getErrorMessages());
+    }
+
+    public function key($key, ...$validators) {
+        foreach($validators as $validator) {
+            $validator($key);
+        }
+
+        return $this;
+    }
+
+    public static function required() {
+        $regexp = '/^(?!\s*$).+/';
+
+        return function($key) use ($regexp) {
+            if (!array_key_exists($key, self::$input)) {
+                self::$errors += [$key => "$key is required"];
+                return;
+            }
+
+            // Array - Must contain at least 1 element
+            if (is_array(self::$input[$key])) {
```

Comment:
Not urgent, since as of now we're only validating against non array values, but for future reference if needed:

To make it easy to test, I would suggest to separate required for array logic. 
So we can start with these simple definitions first:
* Required -> meaning not null & not empty string
* Is Array -> meaning must be an array
* MinSize -> Array min size 

With the above definition we could chain it like this right (using the `->key` implementations)

```
Validator::create()
  ->key('emails', Validator::required(), Validator::isArray(), Validator::minSize(3))
  ->key('apiKey', Validator::required())
```